sklearn/utils/estimator_checks.py

```diff
@@ -1124,6 +1125,41 @@ def check_methods_subset_invariance(name, estimator_orig):
                             atol=1e-7, err_msg=msg)
 
 
+@ignore_warnings(category=FutureWarning)
+def check_methods_sample_order_invariance(name, estimator_orig):
+    # check that method gives invariant results if applied
+    # on a subset with different sample order
+    rnd = np.random.RandomState(0)
+    X = 3 * rnd.uniform(size=(20, 3))
+    X = _pairwise_estimator_convert_X(X, estimator_orig)
+    y = X[:, 0].astype(np.int)
+    if estimator_orig._get_tags()['binary_only']:
+        y[y == 2] = 1
+    estimator = clone(estimator_orig)
+    y = _enforce_estimator_tags_y(estimator, y)
+
+    if hasattr(estimator, "n_components"):
+        estimator.n_components = 1
+    if hasattr(estimator, "n_clusters"):
+        estimator.n_clusters = 1
```

Comment:
Does it make sense to have 1 clusters for this test though? The result is always going to be the same cluster, component so we won't be testing much?

Reply:
umm I'm not entirely sure what `n_clusters` should be instead. Are you suggesting that if the estimator has 2 clusters, then we should test each cluster individually? Apologies if I totally misunderstood your suggestion.

Reply:
No mean that with `n_clusters=1`, for instance `KMeans.predict` whether there is sample invariance or not will predict that one cluster, since there is no alternative values. Could we try changing this to 2 and see if tests still pass?

Reply:
@rth I tried changing to 2 and the tests still pass. I also tried deleting that block of code altogether and the tests also pass. How would you like me to proceed?

Reply:
Please set it to 2.